spark/src/test/java/org/apache/iceberg/spark/source/TestDataSourceOptions.java

```diff
@@ -284,4 +284,103 @@ public void testIncrementalScanOptions() throws IOException {
         .collectAsList();
     Assert.assertEquals("Records should match", expectedRecords.subList(2, 3), result1);
   }
+
+  @Test
+  public void testMetadataSplitSizeOptionOverrideTableProperties() throws IOException {
+    String tableLocation = temp.newFolder("iceberg-table").toString();
+    int splitSize = 2 * 1024;
+
+    HadoopTables tables = new HadoopTables(CONF);
+    PartitionSpec spec = PartitionSpec.unpartitioned();
+    Map<String, String> options = Maps.newHashMap();
+    options.put(TableProperties.SPLIT_SIZE, String.valueOf(128L * 1024 * 1024)); // 128Mb
+    options.put(TableProperties.METADATA_SPLIT_SIZE, String.valueOf(32L * 1024 * 1024)); // 32MB
+    tables.create(SCHEMA, spec, options, tableLocation);
+
+    List<SimpleRecord> expectedRecords = Lists.newArrayList(
+        new SimpleRecord(1, "a"),
+        new SimpleRecord(2, "b")
+    );
+    Dataset<Row> originalDf = spark.createDataFrame(expectedRecords, SimpleRecord.class);
+    originalDf.select("id", "data").write()
+        .format("iceberg")
+        .mode("append")
+        .save(tableLocation);
+
+    int expectedSplits = ((int) tables.load(tableLocation + "#entries")
```

Comment:
nit: I think it would be a bit more readable if we split this logic into multiple lines in all such places. However, it does not matter that much.

```
Table table = tables.create(...);

table.refresh();

List<ManifestFile> manifests = table.currentSnapshot().manifests();
int expectedSplitNum = (manifests.get(0).length() + splitSize -1) / splitSize;
```